chia/util/config.py

```diff
@@ -48,10 +48,24 @@ def config_path_for_filename(root_path: Path, filename: Union[str, Path]) -> Pat
     return root_path / "config" / filename
 
 
-def get_config_lock(root_path: Path, filename: Union[str, Path]) -> BaseFileLock:
+@contextlib.contextmanager
+def lock_config(root_path: Path, filename: Union[str, Path]) -> Iterator[None]:
+    # TODO: This is presently used in some tests to lock the saving of the
+    #       configuration file without having loaded it right there.  This usage
+    #       should probably be removed and this function made private.
     config_path = config_path_for_filename(root_path, filename)
     lock_path: Path = config_path.with_name(config_path.name + ".lock")
-    return FileLock(lock_path)
+    with FileLock(lock_path):
+        yield
+
+
+@contextlib.contextmanager
+def lock_and_load_config(root_path: Path, filename: Union[str, Path]) -> Iterator[Dict[str, Any]]:
+    with lock_config(root_path=root_path, filename=filename):
+        config = _load_config_maybe_locked(root_path=root_path, filename=filename, acquire_lock=False)
+        yield config
+        # TODO: would need to be fancier to integrate this since it is conditional
+        # save_config(root_path=root_path, filename=filename, config_data=config)
```

Comment:
Why is this comment here?

Reply:
Perhaps my use of TODOs is leftover from developing alone and I should move on.  The use is that as people look at the code it is an opportunity to express an added feature of interest that they could be aware of as well as why it is not yet implemented.  Removing.